app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -208,83 +208,107 @@ class HomeActivityTest {
   }
 
   @Test
-  fun testHomeActivity_withAdminProfile_profileNameIsDisplayed() {
+  fun testHomeActivity_hasCorrectActivityLabel() {
+    launch(HomeActivity::class.java).use { scenario ->
+      scenario.onActivity { activity ->
+        val title = activity.title
+        assertThat(title).isEqualTo(context.getString(R.string.home_activity_title))
+      }
+    }
+  }
+
+  @Test
+  fun testHomeActivity_withMorningTimeAndAdminProfile_MorningWithProfileNameIsDisplayed() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(MORNING_TIMESTAMP)
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
       scrollToPosition(position = 0)
       verifyExactTextOnHomeListItemAtPosition(
         itemPosition = 0,
-        targetViewId = R.id.profile_name_text_view,
-        stringToMatch = "Admin!"
+        targetViewId = R.id.welcome_text_view,
+        stringToMatch = "Good morning,Admin!"
       )
     }
   }
 
   @Test
-  fun testHomeActivity_hasCorrectActivityLabel() {
-    launch(HomeActivity::class.java).use { scenario ->
-      scenario.onActivity { activity ->
-        val title = activity.title
-        assertThat(title).isEqualTo(context.getString(R.string.home_activity_title))
-      }
+  fun testHomeActivity_withMorningAndAdminProfile_configChange_MorningWithProfileNameIsDisplayed() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(MORNING_TIMESTAMP)
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+       testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 0)
+      verifyExactTextOnHomeListItemAtPosition(
+        itemPosition = 0,
+        targetViewId = R.id.welcome_text_view,
+        stringToMatch = "Good morning,Admin!"
+      )
     }
   }
 
   @Test
-  fun testHomeActivity_withAdminProfile_configChange_profileNameIsDisplayed() {
+  fun testHomeActivity_withAfternoonTime_withAdminProfile_afternoonWithProfileNameIsDisplayed() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(AFTERNOON_TIMESTAMP)
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
-      onView(isRoot()).perform(orientationLandscape())
```

Comment:
@yashbansal130 Why did you remove this line?

Reply:
> @yashbansal130 Why did you remove this line?

I have added separate tests for each greetings message with and without config changes.